pkg/apis/planetscale/v2/vitessshard_types.go

```diff
@@ -308,26 +308,52 @@ type VitessShardStatus struct {
 	// HasMaster is a condition indicating whether the Vitess topology
 	// reflects a master for this shard.
 	HasMaster corev1.ConditionStatus `json:"hasMaster,omitempty"`
+
 	// HasInitialBackup is a condition indicating whether the initial backup
 	// has been seeded for the shard.
 	HasInitialBackup corev1.ConditionStatus `json:"hasInitialBackup,omitempty"`
-	// MasterAlias is the tablet alias of the master according to the global
-	// shard record. This could be empty either because there is no master,
-	// or because the shard record could not be read. Check the HasMaster
-	// condition whenever the distinction is important.
-	MasterAlias string `json:"masterAlias,omitempty"`
 
 	// Idle is a condition indicating whether the shard can be turned down.
 	// If Idle is True, the shard is not part of the active shard set
 	// (partitioning) for any tablet type in any cell, so it should be safe
 	// to turn down the shard.
 	Idle corev1.ConditionStatus `json:"idle,omitempty"`
 
+	// Conditions is a map of all VitessShard specific conditions we want to set and monitor.
+	Conditions map[VitessShardConditionType]*VitessShardCondition `json:"conditions,omitempty"`
+
+	// MasterAlias is the tablet alias of the master according to the global
+	// shard record. This could be empty either because there is no master,
+	// or because the shard record could not be read. Check the HasMaster
+	// condition whenever the distinction is important.
+	MasterAlias string `json:"masterAlias,omitempty"`
+
 	// BackupLocations reports information about the backups for this shard in
 	// each backup location.
 	BackupLocations []*ShardBackupLocationStatus `json:"backupLocations,omitempty"`
 }
 
+// VitessShardConditionType is a valid value for the key of a VitessShardCondition map where the key is a
+// VitessShardConditionType and the value is a VitessShardCondition.
+type VitessShardConditionType string
+
+// VitessShardCondition contains details for the current condition of this VitessShard.
+type VitessShardCondition struct {
+	// Status is the status of the condition.
+	// Can be True, False, Unknown.
+	// +kubebuilder:validation:Enum=True;False;Unknown
```

Comment:
The values should be comma-separated.

```suggestion
	// +kubebuilder:validation:Enum=True,False,Unknown
```

Reply:
According to the official documentation, they should be semicolon-separated. For reference, their example for `type Alias string`: https://book.kubebuilder.io/reference/generating-crd.html

Reply:
Ahh, this totally different example is comma-separated: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html

Reply:
I'm really confused. Is the first example just flat wrong from their docs?

Reply:
Here's another one that's semicolon-separated. search for `type ConcurrencyPolicy string`: https://book.kubebuilder.io/cronjob-tutorial/api-design.html

Reply:
My guess is they changed the format in a later version of kubebuilder, but we are still using an old version. In any case, you can see from the generated output that semicolon doesn't work with our current generator version. I guess we'll have to keep an eye on this when we upgrade to a newer operator-sdk (and hence pick up a new kubebuilder version) since it looks like a breaking change.

Reply:
Updated https://github.com/planetscale/vitess-operator/pull/31/commits/20698920e71c38d09894ba8b1dfadbe395d879d2.